server/lib/parties/party-service.ts

```diff
@@ -321,18 +324,52 @@ export default class PartyService {
     this.removeClientFromParty(clientSockets, party)
   }
 
-  private clearInviteNotification(partyId: string, user: PartyUser) {
-    this.notificationService
-      .retrieveNotifications({ userId: user.id, type: NotificationType.PartyInvite })
-      .then(async userInviteNotifications => {
-        const notification = userInviteNotifications.filter(n => n.data.partyId === partyId)[0]
-        if (notification) {
-          await this.notificationService.clearById(user.id, notification.id)
-        }
-      })
-      .catch(err => {
-        logger.error({ err }, 'error clearing the invite notification')
+  private async maybeSendInviteNotification(party: PartyRecord, user: PartyUser) {
+    try {
+      const notification = (
+        await this.notificationService.retrieveNotifications({
+          userId: user.id,
+          data: { type: NotificationType.PartyInvite, partyId: party.id },
+          visible: true,
+        })
+      )[0]
+
+      if (notification) {
+        return
+      }
+
+      await this.notificationService.addNotification({
+        userId: user.id,
+        data: {
+          type: NotificationType.PartyInvite,
+          from: party.leader.name,
+          partyId: party.id,
+        },
       })
+    } catch (err) {
+      throw new PartyServiceError(
```

Comment:
If we're gonna drop whatever this error was, probably it should be manually logged. Definitely be careful about handling *all* errors like this, you can easily swallow e.g. syntax errors, problems with database queries, etc. and make it exceptionally hard for us to track down issues. If you know that a particular kind of error can occur and it's fine (e.g. trying to insert something that may violate a unique constraint, but that's a fine problem), you should check explicitly that it is that error.

(honestly this might be a good feature to have in our error payload middleware: an optional Error parameter to log when the response is sent as the "cause" of the handled error, such that it gets linked up with the proper request ID)

Reply:
Yeah, I didn't have a specific error in mind here, just wanted to have a way to let the inviter know in case a party invitation was not sent to the user they invited and to try inviting them again (I added client-side handling of this error in #733).

Good point about swallowing syntax errors and other errors that might indicate something being wrong, so I'm gonna log the error as well.